src/main/java/com/google/solutions/ml/api/vision/BQDestination.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.solutions.ml.api.vision;
+
+import java.io.Serializable;
+import java.util.Objects;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.coders.DefaultCoder;
+
+@DefaultCoder(AvroCoder.class)
```

Comment:
Just a comment: I'm curious, is it kind of standard to use Avro records for serializing data structures as keys (e.g. say, vs simple strings)?

Reply:
Frankly, don't remember why. It was definitely failing with the default serializer with "non-deterministic coder" error. I can re-try if needed.